assets/control-plane/prometheus-rule.yaml

```diff
@@ -15,7 +15,7 @@ spec:
     rules:
     - alert: KubePodCrashLooping
       annotations:
-        description: 'Pod {{ $labels.namespace }}/{{ $labels.pod }} ({{ $labels.container }}) is in waiting state (reason: "CrashLoopBackOff").'
+        description: 'Pod {{ $labels.namespace }}/{{ $labels.pod }} ({{ $labels.container }}) is in waiting state (reason: "CrashLoopBackOff") on cluster {{ $labels.cluster }}.'
```

Comment:
what's our stance on "multi-cluster" rules?

Reply:
Note that I've leaned the other way in the [aforementioned PoW](https://github.com/openshift/cluster-monitoring-operator/pull/2622/files#diff-99c0290799daf9abc6240df64063e20bfaf67b371577b67ac7eec6f4725622ffR117) for TNA since OpenShift's master deck as well as marketing push it as HA. Additionally, TNA is indeed HA based on the defined upstream conditional (for HA).

Reply:
Please note that I've leaned the other way in the [aforementioned PoW](https://github.com/openshift/cluster-monitoring-operator/pull/2622/files#diff-99c0290799daf9abc6240df64063e20bfaf67b371577b67ac7eec6f4725622ffR117) for TNA since OpenShift's master deck as well as marketing push it as HA. Additionally, TNA is indeed HA based on the defined upstream conditional (for HA).

Reply:
Historically, we've opted-into these except for the [Prometheus dashboards](https://github.com/prometheus/prometheus/pull/14062).

Also, KSM, as well as certain parts of control plane mixins (`apiserver`, `windows`, etc.) don't seem to be multi-cluster aware and use `clusterLabel` regardless in their rules (which don't evaluate to anything and are ignored (a `by (foo, cluster)` will resolve to `by (foo)`)).

Reply:
in this particular example, the description would resolve to "[...] is in waiting state (reason: "CrashLoopBackOff") on cluster ", right? It would be confusing for users I assume.

Reply:
I see. I noticed some discrepancies (as mentioned above), but these were limited to the expressions themselves and are non-breaking, which isn't the case for the descriptions. I've therefore reverted midstream's opt-in into multi-cluster control-plane rules (latest commit).